tests/unit/stage/test_cache.py

```diff
@@ -158,3 +162,30 @@ def test_stage_cache_wdir(tmp_dir, dvc, run_copy, mocker):
     assert (tmp_dir / "wdir" / "out_no_cache").exists()
     assert (tmp_dir / "wdir" / "out").read_text() == "out"
     assert (tmp_dir / "wdir" / "out_no_cache").read_text() == "out_no_cache"
+
+
+def test_create_stage_with_params(tmp_dir, dvc, run_copy):
+    p1, p2 = {"foo": 1, "bar": 2}, {"baz": 3, "qux": 4}
+    tmp_dir.gen({"params.yaml": yaml.dump(p1), "myparams.yaml": yaml.dump(p2)})
+    tmp_dir.gen("dep", "dependency")
+    stage = run_copy(
+        "dep",
+        "out",
+        params=["foo,bar", "myparams.yaml:baz,qux"],
+        single_stage=True,
+    )
+    with dvc.state:
+        cache_entry = dvc.stage_cache._load(stage)
```

Comment:
Maybe we should rename `StageCache` to `RunCache`. Just saying.
```suggestion
        cache_entry = dvc.run_cache._load(stage)
```

Reply:
Will do it later when I run into `stage/cache.py`.